### PR TITLE
RfC: add --squash to docker build

### PR DESCRIPTION
**Pull request by MaxPeal**
## Description

first docker build with --squash

## Why is this needed

#2

## How Has This Been Tested?
it needs TESTING and Feeback

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -63,11 +63,23 @@ RUN curl -fL https://github.com/mvdan/sh/releases/download/v2.3.0/shfmt_v2.3.0_l
     chmod +x /bin/shfmt
 
 RUN pip3 install click j2cli packet-python==1.38.2 && \
-    pip3 install coverage
+    pip3 install coverage && \
+    pip3 install docker-squash
```
Comment (mmlb) [inline ci/Dockerfile:67]: Is this still needed if we do the multi-stage builds?

**Comment by jacobweinstock** [comment]
Hey @MaxPeal, I believe there have been some significant changes to the docker builds. Might be helpful to rebase master and checkout the latest diffs to see if this is still something you'd like to see added. 

Also, looks like you will need the DCO sign-off added as well. [ref](https://github.com/tinkerbell/osie/pull/142/checks?check_run_id=1184699307)

**Comment by mmlb** [comment]
#2 was already closed and we have squashed images now (for the images that matter at least). I've got a separate PR for multistage builds #223 that I've tested with buildx which works nicely for multiarch builds + build speed increases but sadly no `squash`. I might grab the `pip3 docker-squash` bit and integrate there though so we can have all the goodies!

**Comment by jacobweinstock** [comment]
Hey @MaxPeal, @mmlb, do we want to update this PR, or would a new PR off master be easier to work with now?

**Comment by nshalman** [comment]
> Hey @MaxPeal, @mmlb, do we want to update this PR, or would a new PR off master be easier to work with now?

After hearing nothing for nearly a month, I'm going to close this out as stale. @MaxPeal or @mmlb please feel free to reopen (or ping me to do so) if that was premature. Thanks folks!

**Pull request closed without merging**